client/src/SolrIndexer.jsx

```diff
@@ -1,25 +1,17 @@
 import React, { useState } from "react";
-import AjaxHelper from "./AjaxHelper";
-
+import { baseUrl } from "./routes";
+import { useFetchContext } from "./context";
 const SolrIndexer = () => {
     const [pid, setPid] = useState("");
     const [results, setResults] = useState("");
+    const {
+        action: { getText },
+    } = useFetchContext();
 
-    function doApiCall(method) {
-        const ajax = AjaxHelper.getInstance();
-        const url = ajax.url + "/messenger/solrindex/" + encodeURIComponent(pid);
-        ajax.ajax({
-            method: method,
-            url: url,
-            dataType: "text",
-            error: function (result, status) {
-                setResults(status);
-            },
-            success: function (result) {
-                setResults(result);
-            },
-        });
-    }
+    const doApiCall = async (method) => {
```

Comment:
Something seems to be wrong with this tool. When I try to preview an index operation, I get:

<pre>
Error: Objects are not valid as a React child (found: [object Promise]). If you meant to render a collection of children, use an array instead.
    in pre (at SolrIndexer.jsx:25)
    in div (at SolrIndexer.jsx:17)
    in SolrIndexer (at Routes.jsx:31)
    in Route (at Routes.jsx:18)
    in Switch (at Routes.jsx:17)
    in Routes (at VuDLPrep.jsx:16)
    in Router (created by BrowserRouter)
    in BrowserRouter (at VuDLPrep.jsx:15)
    in FetchContextProvider (at VuDLPrep.jsx:11)
    in VuDLPrep (at src/index.js:14)
    in StrictMode (at src/index.js:13)
</pre>

Reply:
Made change.

Reply:
forgot to call await